### PR TITLE
WT-5360 Add template for a glossary in developer docs

### PR DESCRIPTION
**Pull request by agorrod**
There is no real content yet, but I thought I'd get it started.

### CODE REVIEW

**Comment by agorrod** [comment]
@tammybailey could you review this for me please? I expect to create a follow on ticket to populate the content.

File: src/docs/devdoc-index.dox [hunk 1]
```diff
@@ -1,5 +1,10 @@
 /*! @page devdoc-index Developer Documentation
 
+@subpage devdoc-glossary
+
+WiredTiger assigns specific meanings to some words, the following page decodes
```
Comment (tammybailey) [inline src/docs/devdoc-index.dox]: Technically, there is no following page as the link is before the text. Maybe something like _"WiredTiger assigns meanings to specific meanings to certain words. Here we decode them."_  Or similar?

File: src/docs/devdoc-glossary.dox [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*! @page devdoc-glossary Developer Documentation Glossary of Terms
+
+WiredTiger has a lot of domain specific nomenclature - this page attempts
+to decode it. This is intended for those navigating the WiredTiger source
+tree - it describes terms internal to the storage engine.
+
+@section devdoc-glossary-general General terms Used in WiredTiger
```
Comment (tammybailey) [inline src/docs/devdoc-glossary.dox]: terms -> Terms

**Comment by agorrod** [comment]
Thanks for the quick code review turnaround @tammybailey

**Pull request merged**